packages/atomic/src/components/common/atomic-focus-trap.tsx

```diff
@@ -1,34 +1,11 @@
 import {Component, Element, Listen, Prop, Watch} from '@stencil/core';
 import {getFirstFocusableDescendant} from '../../utils/accessibility-utils';
-import {defer, getFocusedElement} from '../../utils/utils';
-
-function getParent(element: Element | ShadowRoot) {
-  if (element.parentNode) {
-    return element.parentNode as Element | ShadowRoot;
-  }
-  if (element instanceof ShadowRoot) {
-    return element.host;
-  }
-  return null;
-}
-
-function contains(
-  ancestor: Element | ShadowRoot,
-  element: Element | ShadowRoot
-): boolean {
-  if (element === ancestor) {
-    return true;
-  }
-  if (
-    element instanceof HTMLElement &&
-    element.assignedSlot &&
-    contains(ancestor, element.assignedSlot)
-  ) {
-    return true;
-  }
-  const parent = getParent(element);
-  return parent === null ? false : contains(ancestor, parent);
-}
+import {
+  defer,
+  getFocusedElement,
+  getParent,
+  isAncestorOf,
+} from '../../utils/utils';
```

Comment:
for anyone wondering: this was copied from the changes I need to make for the search box in order to implement `atomic-focus-detector`. I think it's fine for this change to stay in this PR.